src/batch_builder.rs

```diff
@@ -1884,12 +1884,19 @@ impl BoxShadowMetrics {
 
 pub fn compute_box_shadow_rect(box_bounds: &Rect<f32>,
                                box_offset: &Point2D<f32>,
-                               spread_radius: f32)
+                               spread_radius: f32,
+                               clip_mode: BoxShadowClipMode)
                                -> Rect<f32> {
     let mut rect = (*box_bounds).clone();
     rect.origin.x += box_offset.x;
     rect.origin.y += box_offset.y;
-    rect.inflate(spread_radius, spread_radius)
+
+    let _spread = match clip_mode {
```

Comment:
I would just make this:

    if clip_mode == BoxShadowClipMode::Inset {
        spread_radius = -spread_radius
    }

and change `spread_radius` in the function signature above to `mut spread_radius`.